src/OpenSpaceToolkit/Astrodynamics/NumericalSolver.cpp

```diff
@@ -0,0 +1,407 @@
+                    case NumericalSolver::StepperType::RungeKuttaCashKarp54:
+                        integrate_const (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_54() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                    case NumericalSolver::StepperType::RungeKuttaFehlberg78:
+                        integrate_const (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_78() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                }
+        }
+        case NumericalSolver::LogType::LogAdaptive:
+        {
+            switch (stepperType_)
+                {
+                    case NumericalSolver::StepperType::RungeKuttaCashKarp54:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_54() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                    case NumericalSolver::StepperType::RungeKuttaFehlberg78:
+                        integrate_adaptive (make_controlled( absoluteTolerance_, relativeTolerance_, NumericalSolver::error_stepper_type_78() ), aSystemOfEquations, aStateVector, (0.0), propDurationInSecs, adjustedTimeStep, NumericalSolver::NumericalIntegrationLogger) ;
+                        return aStateVector ;
+                }
+        }
+    }
+
+    throw ostk::core::error::runtime::Undefined("No StateVector returned from Odeint") ;
+
+    return NumericalSolver::StateVector(anInitialStateVector.size()) ;
+
+}
+
+
+NumericalSolver::StateVector    NumericalSolver::integrateStateFromInstantToInstant (   const   StateVector&        anInitialStateVector,
+                                                                                        const   Instant&            aStartInstant,
+                                                                                        const   Instant&            anEndInstant,
+                                                                                        const   NumericalSolver::SystemOfEquationsWrapper&  aSystemOfEquations  ) const
+{
+
+// [TBI] Incldue safety checks to make sure incoming parameters don't break stuff
+
+    NumericalSolver::StateVector aStateVector = anInitialStateVector ;
+
+    double propDurationInSecs ;
+    if ( ((anEndInstant-aStartInstant).inSeconds()).isZero() ) // If integration duration is zero seconds long, skip integration
+    {
+        return anInitialStateVector ;
+    }
+    else
+    {
+        propDurationInSecs = (double)((anEndInstant-aStartInstant).inSeconds()) ;
```

Comment:
```suggestion
        propDurationInSecs = (double)((anEndInstant - aStartInstant).inSeconds()) ;
```

on top of other changes

Reply:
fixed